integration/plugin_test.go

```diff
@@ -0,0 +1,111 @@
+//go:build integration
+
+package integration
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
+)
+
+func TestPlugin(t *testing.T) {
+	tests := []struct {
+		name       string
+		plugin     string
+		pluginArgs string
+		golden     string
+	}{
+		{
+			name:   "count plugin installed from `index`",
+			plugin: "count@v0.2.0",
+			golden: "testdata/count-0.2.0-plugin.txt.golden",
+		},
+		{
+			name:       "count plugin installed from github archive",
+			plugin:     "https://github.com/aquasecurity/trivy-plugin-count/archive/refs/tags/v0.1.0.zip",
+			pluginArgs: "--published-before=2020-01-01",
+			golden:     "testdata/count-0.1.0-plugin-with-before-flag.txt.golden",
+		},
+	}
+
+	// Set up testing DB
+	cacheDir := initDB(t)
+	tempStdOut := setTempStdout(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// We can overwrite stdout for `_default_Manager` only once.
+			// So we need to clear the temporary stdout file before each test case.
+			clearFile(t, tempStdOut)
+
+			t.Setenv("XDG_DATA_HOME", t.TempDir())
+
+			// Install plugin
+			err := execute([]string{
+				"plugin",
+				"install",
+				tt.plugin,
+			})
+			require.NoError(t, err)
+
+			// Get list of plugins
+			err = execute([]string{
+				"plugin",
+				"list",
+			})
+			require.NoError(t, err)
+
+			// Run Trivy with plugin as output
+			args := []string{
+				"--cache-dir",
+				cacheDir,
+				"fs",
+				"-f",
+				"json",
+				"-o",
+				"plugin=count",
+				"testdata/fixtures/repo/pip",
+			}
+
+			if tt.pluginArgs != "" {
+				args = append(args, "--output-plugin-arg", tt.pluginArgs)
+			}
+
+			err = execute(args)
+
+			if *update {
+				fsutils.CopyFile(tempStdOut.Name(), tt.golden)
+			}
+
+			compareRawFiles(t, tt.golden, tempStdOut.Name())
+		})
+	}
+}
+
+func setTempStdout(t *testing.T) *os.File {
+	tmpFile := filepath.Join(t.TempDir(), "output.txt")
+	f, err := os.Create(tmpFile)
+	require.NoError(t, err)
+
+	// Overwrite Stdout to get output of plugin
+	defaultStdout := os.Stdout
+	os.Stdout = f
```

Comment:
What if passing io.Writer here?
https://github.com/aquasecurity/trivy/blob/88ba46047c93e6046292523ae701de774dfdc4dc/integration/integration_test.go#L284

Reply:
I mean 
```
func execute(osArgs []string, out io.Writer) error {
    if out == nil {
        out = io.Discard
    }
    ...
}

Reply:
I tried to use your idea.
But we use `os.Stdout` in `plugin` package.

I used [NewManager](https://github.com/aquasecurity/trivy/blob/0c6687d5ba0df5de2632aad26c21ea99565bd9e1/pkg/plugin/manager.go#L60) instead of [defaultManager](https://github.com/aquasecurity/trivy/blob/0c6687d5ba0df5de2632aad26c21ea99565bd9e1/pkg/plugin/manager.go#L75) in `app.go` to use `Stdout` from `cobra`.
It works, but there is problem with case when plugin is output.
We need to add output for plugin in Options:
https://github.com/aquasecurity/trivy/blob/0c6687d5ba0df5de2632aad26c21ea99565bd9e1/pkg/flag/options.go#L511-L533

i am not sure that we need to do that now.
I would leave this workaround until the tests become more numerous and complex, which would require these changes.

But if you want to do it - tell me and I will continue the research

Reply:
OK, we can improve it later.